src/Model/CodeModelGo.cs

```diff
@@ -277,18 +287,40 @@ internal void CreateFutureTypeForMethod(MethodGo method)
                 throw new InvalidOperationException("CreateFutureTypeForMethod requires method to be a long-running operation");
             }
 
-            // don't create duplicate future types
-            var future = new FutureTypeGo(method);
-            if (_futureTypes.ContainsKey(future))
+            // this is the future to return from the method
+            var future = GetOrAddFuture(new FutureTypeGo(method));
+
+            // if this is a pageable method create a future type for the
+            // "list all" method wrapped in our custom response type
+            if (method.IsPageable)
             {
-                future = _futureTypes[future];
+                var listAllFuture = GetOrAddFuture(new FutureTypeGo(CodeNamerGo.Instance.GetFutureTypeName($"{method.Group}{method.Name}All"), method));
+                method.ReturnType = new LroPagedResponseGo(future, listAllFuture, method.ReturnType.Headers);
             }
             else
             {
-                _futureTypes.Add(future, future);
+                method.ReturnType = new Response(future, method.ReturnType.Headers);
             }
+        }
 
-            method.ReturnType = new Response(future, method.ReturnType.Headers);
+        /// <summary>
+        /// Checks if the specified future type already exists, if it does return that one instead.
+        /// If it does not exist it is added to the collection of model types and returned.
+        /// </summary>
+        /// <param name="futureType">The future type to check for and possibly add.</param>
+        /// <returns>The existing or added object.</returns>
+        private FutureTypeGo GetOrAddFuture(FutureTypeGo futureType)
+        {
+            // don't create duplicate future types
+            if (ModelTypes.Contains(futureType))

```

Comment:
>ModelTypes.Contains(futureType)) [](start = 16, length = 32)

I presume that the `FutureTypeGo`  was updated to check for something else except the object reference ? ( Since i see all the uses of this method are adding new objects )